### DISCUSSION TITLE
Converted hardcoded HTML to JSON

### DISCUSSION THREAD

**Discussion by 4rnv** [Buzzpy/Dev-Encyclopedia#29] [Ideas]
Hello,
I wrote a [python script](https://github.com/4rnv/HTML-to-JSON) which converts the HTML cards to JSON. The card info for this project is in the [data.json](https://github.com/4rnv/HTML-to-JSON/blob/main/data.json) file in the repo. The cards can then be generated dynamically using Javascript. I tested it on localhost and it works perfectly. This would make it easier to add new cards to the encyclopedia, eliminating the need to repeatedly write HTML. Although if this site gets really big it would be better to make a proper database.

**Comment by bartosz-skejcik** [comment]
I think thats a great idea but even better one would be to go one step further and transition into astro.js and store the data for each category in markdown files for easier editing experience. we can dinamicaly get the onclick from the name of the card in the md file

The quicker we move to a framework the simpler it will be to transition. Further down the road it will be much harder cause of the complexity of components routes etc.

**Comment by Richard-Woessner** [comment]
I saw some PR for this, did Buzzby changer her mind and not want it to be Json driven

**Reply by Buzzpy** [reply]
Actually, I integrated Astro and JSON files! @4rnv, I too used a similar script, and if you take a look at [astro-devpedia.pages.dev/](https://astro-devpedia.pages.dev/) you can see the JSON-driven website alive!

We'll be merging it to the main branch as soon as possible.

**Reply by 4rnv** [reply]
> I saw some PR for this, did Buzzby changer her mind and not want it to be Json driven

It is JSON based now! Each card has its own JSON file on the astro-integration branch. It's in the content directory.
